### PR TITLE
Add SQL activity log

### LINKED ISSUE #111: Implement Activity Log section
Every time the user runs a query we should store and display as a list in a section under the schema explorer:
- If the query was successful with a ✅ or X
- Latency.
- Copy SQL.

### PR DESCRIPTION
**Pull request by bobbyiliev**
As per #111 adding SQL activity log which contains:

- ✅  or ❌  depending if the query was successful or not
- The time it took the query to complete
- Queries show from newest to oldest
- When clicking on a query from the log it copies it in your clipboard

activity-log-sql
    
 Closes #111

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -106,10 +121,21 @@ export function activate(vsContext: vscode.ExtensionContext) {
 
     vsContext.subscriptions.push(runDisposable);
     vsContext.subscriptions.push(copyDisposable);
+
+    let copySQLDisposable = vscode.commands.registerCommand('extension.copySQL', (sql: string) => {
+        vscode.env.clipboard.writeText(sql).then(() => {
+            vscode.window.showInformationMessage('SQL copied to clipboard!');
```
Comment (joacoc) [inline src/extension.ts:127]: I initially loved to use these notifications, but I suddenly fell into the trap and notifications were spawning everywhere, so I relied on the VS Code guidelines for notifications and is really neat and most of the cases ends in the relax 😎 https://code.visualstudio.com/assets/api/ux-guidelines/examples/notification-decision-tree.png

**Review by joacoc — APPROVED** [review]
LGTM. Left some comments about UX/UI guidelines by VS Code, and other comment to add support to the Data Storage.

File: src/extension.ts (hunk 1 above)
Comment (bobbyiliev) [reply src/extension.ts]: Good point, just removed it.

File: src/providers/activity.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import * as vscode from 'vscode';
+import Context, { EventType } from '../context/context';
+import path = require('path');
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply src/providers/activity.ts]: Seems like I forgot to remove it

File: src/providers/activity.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+export default class ActivityLogTreeProvider implements vscode.TreeDataProvider<ActivityLogNode> {
+
+    private _onDidChangeTreeData: vscode.EventEmitter<ActivityLogNode | undefined | null | void> = new vscode.EventEmitter<ActivityLogNode | undefined | null | void>();
+    readonly onDidChangeTreeData: vscode.Event<ActivityLogNode | undefined | null | void> = this._onDidChangeTreeData.event;
+
+    private logs: ActivityLog[] = [];
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply src/providers/activity.ts:8]: Good idea, just extended this to use the data storage!

File: src/providers/activity.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    status: "success" | "failure";
+    latency: number;
+    sql: string;
+}
+
+class ActivityLogItem extends vscode.TreeItem {
+    constructor(public readonly log: ActivityLog) {
+        // Shorten the displayed query if it's too long
+        const shortSQL = log.sql.length > 50 ? log.sql.substring(0, 50) + "..." : log.sql;
+        super(`${log.status === "success" ? "✅" : "❌"} ${shortSQL}`, vscode.TreeItemCollapsibleState.None);
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply src/providers/activity.ts]: Ah yes this makes sense! Just changed the emojies to icons from the figma project above!

image

File: src/providers/activity.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        const shortSQL = log.sql.length > 50 ? log.sql.substring(0, 50) + "..." : log.sql;
+
+        super(shortSQL, vscode.TreeItemCollapsibleState.None);
+
+        // Set iconPath based on the status
+        const iconName = log.status === "success" ? "success_icon.svg" : "error_icon.svg";
+        this.iconPath = vscode.Uri.file(path.join(__dirname, '..', 'resources', iconName));
+
+        // Update the description to display latency with a stopwatch emoji.
+        this.description = `⏱️ ${log.latency}ms`;
```
Comment (joacoc) [inline src/providers/activity.ts:58]:
```suggestion
        this.description = ` ${log.latency}ms`;
```

**Review by joacoc — COMMENTED** [review]
LGTM 🚀 - Removing the emoji to keep consistency and avoid repetition with the other parts. Feel free to merge!

**Pull request merged**